sdk/core/src/headers/mod.rs

```diff
@@ -1,57 +1,57 @@
 mod utilities;
 
 use http::request::Builder;
+
+pub use hyper::header::{IF_MODIFIED_SINCE, USER_AGENT};
 pub use utilities::*;
 
 pub const MS_DATE: &str = "x-ms-date";
 
+pub trait AddAsHeader {
+    fn add_as_header(&self, builder: Builder) -> Builder;
+}
+
 #[must_use]
-pub fn add_optional_header_ref<T: crate::AddAsHeader>(
-    item: &Option<&T>,
-    mut builder: Builder,
-) -> Builder {
+pub fn add_optional_header_ref<T: AddAsHeader>(item: &Option<&T>, mut builder: Builder) -> Builder {
     if let Some(item) = item {
         builder = item.add_as_header(builder);
     }
     builder
 }
 
 #[must_use]
-pub fn add_optional_header<T: crate::AddAsHeader>(
-    item: &Option<T>,
-    mut builder: Builder,
-) -> Builder {
+pub fn add_optional_header<T: AddAsHeader>(item: &Option<T>, mut builder: Builder) -> Builder {
     if let Some(item) = item {
         builder = item.add_as_header(builder);
     }
     builder
 }
 
 #[must_use]
-pub fn add_mandatory_header<T: crate::AddAsHeader>(item: &T, builder: Builder) -> Builder {
+pub fn add_mandatory_header<T: AddAsHeader>(item: &T, builder: Builder) -> Builder {
     item.add_as_header(builder)
 }
 
-pub const SERVER: &str = "server"; // -> [String]
+pub const SERVER: &str = "server";
 pub const SOURCE_IF_MODIFIED_SINCE: &str = "x-ms-source-if-modified-since";
```

Comment:
I wonder if this approach still makes sense. It's vey C to have constants defined in a separate file because file embedding is common. In Rust this creates a vicious coupling: it makes necessary to modify the core crate whenever a new header is defined elsewhere. 

Maybe it's better to simply use the constant whenever it's used instead. For example, reading `SOURCE_IF_MODIFIED_SINCE` instead of `x-ms-source-if-modified-since` does not give much more information

What do you think?

Reply:
I believe most if not all of these headers are used across services so they make sense to have in the core crate. However, it could make sense to move the constants into the `request_options` module to the specific places where they are used.